### PR TITLE
[ViewBinding]-ankiPackage

### PR DESCRIPTION
**Pull request by Jatin887**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
Part of #11116

## Approach
_How does this change address the problem?_

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration (SDK version(s), emulator or physical, etc)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Jatin887** [comment]
@david-allison i converted all the activities and fragments.
I have one doubt do you want me to create viewBinding classes for progress bars and toolbars? Because only that part is left

File: AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt [hunk 1]
```diff
@@ -114,13 +114,11 @@ class CardTemplateBrowserAppearanceEditor : AnkiActivity() {
     }
 
     private fun initializeUiFromBundle(bundle: Bundle) {
-        setContentView(R.layout.card_browser_appearance)
+        binding = CardBrowserAppearanceBinding.inflate(layoutInflater)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardTemplateBrowserAppearanceEditor.kt:117]: can this be moved to `onCreate`?

File: AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt [hunk 2]
```diff
@@ -34,17 +35,19 @@ import java.io.FileNotFoundException
  * To access this screen: Add/Edit Note - Attachment - Add Image - Drawing
  */
 class DrawingActivity : AnkiActivity() {
-    private lateinit var mColorPalette: LinearLayout
+    private lateinit var binding: ActivityDrawingBinding
+    private lateinit var bindingReviewerWhiteboardEditorBinding: ReviewerWhiteboardEditorBinding
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt:39]: This needs shortening, maybe `whiteboardBinding`?

File: AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt [hunk 3]
```diff
@@ -62,10 +65,10 @@ class DrawingActivity : AnkiActivity() {
             }
             R.id.action_whiteboard_edit -> {
                 Timber.i("Drawing:: Pen Color button pressed")
-                if (mColorPalette.visibility == View.GONE) {
-                    mColorPalette.visibility = View.VISIBLE
+                if (bindingReviewerWhiteboardEditorBinding.whiteboardEditor.visibility == View.GONE) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt:68]: Is this only a color palette?

This one might want a rename

File: AnkiDroid/src/main/java/com/ichi2/anki/Info.kt [hunk 4]
```diff
@@ -60,56 +63,65 @@ class Info : AnkiActivity() {
...
+                    progressBarBinding.progressBar.visibility = View.GONE
                 }
             }
         }
-        val marketButton = findViewById<Button>(R.id.left_button)
         if (canOpenMarketUri()) {
-            marketButton.setText(R.string.info_rate)
-            marketButton.setOnClickListener { tryOpenIntent(this, AnkiDroidApp.getMarketIntent(this)) }
+            binding.leftButton.setText(R.string.info_rate)
+            binding.leftButton.setOnClickListener {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Info.kt:83]: This needs either a refactoring or fixing, `marketButton` is much more readable

File: AnkiDroid/src/main/java/com/ichi2/anki/Info.kt [hunk 5]
```diff
@@ -60,56 +63,65 @@ class Info : AnkiActivity() {
...
-                val continueButton = findViewById<Button>(R.id.right_button)
-                continueButton.text = res.getString(R.string.dialog_continue)
-                continueButton.setOnClickListener { close() }
+                binding.rightButton.text = res.getString(R.string.dialog_continue)
+                binding.rightButton.setOnClickListener { close() }
                 val background = String.format("#%06X", 0xFFFFFF and backgroundColor)
-                mWebView!!.loadUrl("file:///android_asset/changelog.html")
-                mWebView!!.settings.javaScriptEnabled = true
-                mWebView!!.webViewClient = object : WebViewClient() {
+                binding.info.loadUrl("file:///android_asset/changelog.html")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Info.kt:116]: again: `webView` provides more information on the type

File: AnkiDroid/src/main/java/com/ichi2/anki/Info.kt [hunk 6]
```diff
@@ -178,7 +195,13 @@ class Info : AnkiActivity() {
             val content = res.getStringArray(R.array.about_content)
             append("<html><style>body {color:").append(textColor).append(";}</style>")
             append("<body text=\"#000000\" link=\"#E37068\" alink=\"#E37068\" vlink=\"#E37068\">")
-            append(String.format(content[0], res.getString(R.string.app_name), res.getString(R.string.link_anki)))
+            append(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Info.kt:198]: changed without a reason

File: AnkiDroid/src/main/java/com/ichi2/anki/Info.kt [hunk 7]
```diff
@@ -150,16 +162,20 @@ class Info : AnkiActivity() {
         val clipboardManager = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager?
         if (clipboardManager != null) {
             clipboardManager.setPrimaryClip(ClipData.newPlainText(this.title, debugInfo))
-            showThemedToast(this, getString(R.string.about_ankidroid_successfully_copied_debug), true)
+            showThemedToast(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Info.kt:165]: changed without a reason

File: AnkiDroid/src/main/java/com/ichi2/anki/Info.kt [hunk 8]
```diff
@@ -194,7 +217,12 @@ class Info : AnkiActivity() {
                     res.getString(R.string.link_contribution)
                 )
             ).append(" ")
-            append(String.format(content[3], res.getString(R.string.link_translation))).append("<br/><br/>")
+            append(
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/Info.kt:220]: ditto

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt [hunk 9]
```diff
@@ -404,7 +405,7 @@ class ModelBrowser : AnkiActivity() {
                         mModels!![mModelListPosition].put("name", deckName)
                         mModelDisplayList!![mModelListPosition] = DisplayPair(
                             mModels!![mModelListPosition].getString("name"),
-                            mCardCounts!![mModelListPosition].toInt()
+                            mCardCounts!![mModelListPosition]
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt:408]: was this change intended? Seems good, but not part of the PR's scope

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 10]
```diff
@@ -193,26 +173,20 @@ class MyAccount : AnkiActivity() {
             }
         )
 
-        val loginButton = mLoginToMyAccountView.findViewById<Button>(R.id.login_button)
-        loginButton.setOnClickListener { login() }
-        val resetPWButton = mLoginToMyAccountView.findViewById<Button>(R.id.reset_password_button)
-        resetPWButton.setOnClickListener { resetPassword() }
-        val signUpButton = mLoginToMyAccountView.findViewById<Button>(R.id.sign_up_button)
+        binding.loginButton.setOnClickListener { login() }
+        binding.resetPasswordButton.setOnClickListener { resetPassword() }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:177]: this change is awesome!

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 11]
```diff
@@ -242,7 +216,7 @@ class MyAccount : AnkiActivity() {
     /**
      * Listeners
      */
-    val mLoginListener: Connection.TaskListener = object : Connection.TaskListener {
+    private val mLoginListener: Connection.TaskListener = object : Connection.TaskListener {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:219]: this is a refactoring and shouldn't be included in a 'viewbinding' change

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 12]
```diff
@@ -294,7 +268,7 @@ class MyAccount : AnkiActivity() {
         }
     }
 
-    protected fun getHumanReadableLoginErrorMessage(exception: Exception?): String? {
+    private fun getHumanReadableLoginErrorMessage(exception: Exception?): String? {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:271]: ditto

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt (hunk 3 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/DrawingActivity.kt:68]: Yes it is i will shorten the name

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt (hunk 9 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelBrowser.kt:408]: no i just solve that. so these files should not be fall into Kotlin Cleanup issue :)

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt (hunk 12 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt]: done

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt (hunk 11 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt]: done

**Comment by Jatin887** [comment]
@david-allison added-changes

**Pull request closed without merging**